src/diracx/routers/job_manager/__init__.py

```diff
@@ -233,14 +236,101 @@ def __init__(self, user_info: UserInfo, allInfo: bool = True):
 
 
 @router.delete("/")
-async def delete_bulk_jobs(job_ids: Annotated[list[int], Query()]):
+async def delete_bulk_jobs(
+    job_ids: Annotated[list[int], Query()],
+    config: Annotated[Config, Depends(ConfigSource.create)],
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    # TODO: implement job policy
+    try:
+        await asyncio.gather(
+            *(
+                delete_job(
+                    job_id,
+                    config,
+                    job_db,
+                    job_logging_db,
+                    task_queue_db,
+                    background_task,
+                )
+                for job_id in job_ids
+            )
+        )
```

Comment:
Would it be practical to make this an actual bulk operation on the DB side? i.e. something like:

```
DELETE FROM JOBS WHERE JOB_ID IN (123, 124, 125, 126)
```

rather than

```
DELETE FROM JOBS WHERE JOB_ID = 123
DELETE FROM JOBS WHERE JOB_ID = 124
DELETE FROM JOBS WHERE JOB_ID = 125
DELETE FROM JOBS WHERE JOB_ID = 126
```

(This is probably better as an issue and then a separate PR rather than adding complexity to this one.)

Reply:
I ended up doing it here because I had problems using the [TaskGroup](https://docs.python.org/3/library/asyncio-task.html#asyncio.TaskGroup)+except* as you advised, and I wanted to see if making bulk methods could help me avoid theses. In the end, I still had to use them, but most of the work was already done. Sorry for the complexity.